### PR TITLE
Add ability to add/change/delete notes and search by note content. Modify error handling

### PR DESCRIPTION
**Pull request by TetianaFilonenko**
- Add classes for `note` and `note_book` and functions to add/change/remove notes 
- Update console bot with a new commands 
- Use more generic error handling
- Search by note content
- Validate note description is present
image
Search 
image

### CODE REVIEW

File: console_bot/errors.py [hunk 1]
```diff
@@ -1,21 +1,12 @@
...
-    """Contact has not been found"""
-
 def input_error(func):
+    # FIXME: message here are too specific
     def inner(*args, **kwargs):
         try:
             return func(*args, **kwargs)
         except ValueError:
-            return "Give me name and phone please."
+            return "Invalid input. Please provide valid arguments."
```
Comment (svilav) [inline console_bot/errors.py:7]: 👍

File: console_bot.py [hunk 2]
```diff
@@ -14,6 +14,11 @@ def print_help():
   add-birthday [name] [birthday] - Adds birthday to the contact
   show-birthday [name]           - Shows birthday for specific contact
   birthdays                      - Shows birthdays for all contacts celebrating next week
+  add-note
+  find-notes
+  delete-note
+  edit-note
+  all-notes
```
Comment (svilav) [inline console_bot.py]: Don't forget to add descriptions here

File: console_bot/note.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        """Function to delete a note at a specific index."""
+        if 0 <= index < len(self.data):
+            del self.data[index]
+            return "Note was deleted"
+        else:
+            raise IndexError("Index out of range.")
+
+    def __str__(self):
+        """Convert notebook to string."""
+        return '\n'.join(f"{index}: {note}" for index, note in enumerate(self.data))
```
Comment (svilav) [inline console_bot/note.py]: 👍

File: console_bot/note.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+"""Module providing classes and functions for working with notes"""
+from collections import UserList
+
+class Note:
+    """Class representing a note"""
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
```
Comment (svilav) [inline console_bot/note.py]: newline here

File: console_bot/note.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+"""Module providing classes and functions for working with notes"""
+from collections import UserList
+
+class Note:
```
Comment (svilav) [inline console_bot/note.py]: 2 newlines here

File: console_bot/note.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        return self.__value
+
+    @value.setter
+    def value(self, new_value):
+        if new_value.strip() == "":
+            raise ValueError("New value cannot be None or empty")
+        self.__value = new_value
+
+
+    def to_dict(self):
```
Comment (svilav) [inline console_bot/note.py]: here needs to be 1 newline

**Review by svilav — APPROVED** [review]
LGTM. But I strongly suggest to add styling extension to VS Code as we have PEP requirement in the project

**Comment by svilav** [comment]
I tried it, works well 👍

**Comment by TetianaFilonenko** [comment]
@svilav What styling plugin do you use?

File: console_bot.py (hunk 2 above)
Comment (TetianaFilonenko) [reply console_bot.py]: Good point! I have some idea about showing help message, will introduce in separate PR and update the description

**Pull request merged**